sqlglot/dialects/tsql.py

```diff
@@ -667,7 +667,19 @@ def create_sql(self, expression: exp.Create) -> str:
                 if kind == "SCHEMA":
                     sql = f"""IF NOT EXISTS (SELECT * FROM information_schema.schemata WHERE schema_name = {identifier}) EXEC('{sql}')"""
                 elif kind == "TABLE":
-                    sql = f"""IF NOT EXISTS (SELECT * FROM information_schema.tables WHERE table_name = {identifier}) EXEC('{sql}')"""
+                    assert table
+                    where = f"table_name = {self.sql(exp.Literal.string(table.name))}"
```

Comment:
I don't think the `exp.Literal.string` adds anything here, you can simply do `f"table_name = {table.name}"`.